File: q3/code.js
var lengthOfLongestSubstring = function(s) {
    var i = 0,
        l = s.length,
        m = [],
        x = 0,
        target = 0,
        offset = 0,
        maxlen = 0,
        len = 0;

    for(; i < l|0; ++i) {
        target = s[i].codePointAt()|0;
        x = m[target]|0;
        m[target] = i + 1;

        if(x > offset) { offset = x|0; }

        len = i - offset + 1;
        if (maxlen < len) { maxlen = len|0; }
    }

    return maxlen;
};

File: q3/code.go
package q3

func lengthOfLongestSubstring(s string) int {
	m := make([]int, 127) // ascii code mapping.
	offset := 0
	maxlen := 0

	for i, target := range s {
		x := m[target]
		if x > offset {
			offset = x
		}
		m[target] = i + 1
		if maxlen < i - offset + 1 {
			maxlen = i - offset + 1
		}
	}
	return maxlen
}
